packages/go_router/tool/run_tests.dart

```diff
@@ -9,25 +9,103 @@
 // ignore_for_file: avoid_print
 
 import 'dart:io';
+import 'package:io/io.dart' as io;
 import 'package:path/path.dart' as p;
+import 'package:yaml_edit/yaml_edit.dart';
 
 Future<void> main(List<String> args) async {
-  if (!Platform.isMacOS) {
-    print('This test can only be run on macOS.');
+  if (!Platform.isMacOS && !Platform.isWindows) {
+    print('This test can only be run on macOS or windows.');
     exit(0);
   }
+  final p.Context ctx = p.context;
   final Directory packageRoot =
-      Directory(p.dirname(Platform.script.path)).parent;
+      Directory(ctx.dirname(_ensureTrimLeadingSeparator(Platform.script.path)))
+          .parent;
+
+  // Copy test_fixes/ to be tested in a temp directory.
+  // This ensures the dart fix can be applied to projects
+  // outside of this package.
+  final Directory testFixesTargetDir = await Directory.systemTemp.createTemp();
+
+  await _prepareTemplate(
+    ctx: ctx,
+    testFixesDir: ctx.join(packageRoot.path, 'test_fixes'),
+    testFixesTargetDir: testFixesTargetDir.path,
+  );
+  final int pubGet = await _runProcess(
+    'dart',
+    <String>[
+      'pub',
+      'upgrade',
+    ],
+    workingDirectory: testFixesTargetDir.path,
+  );
+  if (pubGet != 0) {
+    await cleanUpAndExit(statusCode: pubGet, toDelete: testFixesTargetDir);
+  }
   final int status = await _runProcess(
     'dart',
     <String>[
       'fix',
       '--compare-to-golden',
     ],
-    workingDirectory: p.join(packageRoot.path, 'test_fixes'),
+    workingDirectory: testFixesTargetDir.path,
   );
+  if (status != 0) {
+    await cleanUpAndExit(statusCode: status, toDelete: testFixesTargetDir);
+  }
+  await cleanUpAndExit(statusCode: 0, toDelete: testFixesTargetDir);
+}
 
-  exit(status);
+Future<Never> cleanUpAndExit({
+  required int statusCode,
+  required Directory toDelete,
+}) async {
+  await toDelete.delete(recursive: true);
+  exit(statusCode);
+}
+
+Future<void> _prepareTemplate({
+  required String testFixesDir,
+  required String testFixesTargetDir,
+  required p.Context ctx,
+}) async {
+  await io.copyPath(testFixesDir, testFixesTargetDir);
+
+  final String pubspecYamlPath = ctx.join(testFixesTargetDir, 'pubspec.yaml');
+  final File targetPubspecPath = File(pubspecYamlPath);
+  const String initialYaml = '''
+name: test_fixes
+publish_to: "none"
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.18.0 <4.0.0"
+  flutter: ">=3.3.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  go_router:
+    path:
+''';
+  final YamlEditor editor = YamlEditor(initialYaml);
+  editor.update(
+    <String>['dependencies', 'go_router', 'path'],
+    ctx.dirname(testFixesDir),
+  );
+  final String newYaml = editor.toString();
+  await targetPubspecPath.writeAsString(newYaml);
+}
+
+String _ensureTrimLeadingSeparator(String path) {
```

Comment:
The name of this function is very misleading, since it doesn't remove the leading `/` on any platform except Windows, and even then `/` isn't the default path separator on Windows.

It's not clear to me what the use case for this is; i.e., why we have a path on windows that starts with a `/` in the first place. This really needs a comment explaining what it's for, and a clearer name (I'm not sure what to suggest offhand since I don't understand the conceptual intent of the method).